WordPress/Classes/Models/Notifications/Types/NewPostNotification.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+struct NewPostNotification {
+
+    // MARK: - Properties
+
+    let note: Notification
+    let postID: UInt
+    let siteID: UInt
+
+    var liked: Bool {
+        get {
+            getPostLikedStatus()
+        } set {
+            updatePostLikedStatus(newValue)
+        }
+    }
+
+    // MARK: - Init
+
+    init?(note: Notification) {
+        guard let postID = note.metaPostID?.uintValue, let siteID = note.metaSiteID?.uintValue else {
```

Comment:
Can you provide an example?

Reply:
So when this init return nil `URL(string:)`, it is obvious to why that would be the case. But for this one, `Notification` being a very large and complicated entity, it would be very implicit why the init fails.

My point is, the caller would have to read the implementation of the init to find out why init returns nil.

I think the proper way of addressing this would be passing exactly what the struct needs rather than passing the whole `Notification`. So instead of doing `note.body(ofType`, the struct can accept a closure that returns `[String: Any]?` and an `updateClosure`.